gateway-service/src/main/java/org/zowe/apiml/gateway/conformance/ValidateAPIController.java

```diff
@@ -69,33 +73,62 @@ public class ValidateAPIController {
+            instanceCheck(serviceInstances);
 
-        ServiceInstance serviceInstance = serviceInstances.get(0);
-        Map<String, String> metadata = getMetadata(serviceInstance);
+            ServiceInstance serviceInstance = serviceInstances.get(0);
+            Map<String, String> metadata = getMetadata(serviceInstance);
 
-        foundNonConformanceIssues.put(METADATA_PROBLEMS, metaDataCheck(metadata));
-        if (foundNonConformanceIssues.size() != 0) {     // cant continue without metadata
-            return generateBadRequestResponseEntity(NO_METADATA_KEY, foundNonConformanceIssues);
+            metaDataCheck(metadata);
+
+            String swaggerUrl = verificationOnboardService.findSwaggerUrl(metadata);
+
+            validateSwaggerDocument(serviceId, foundNonConformanceIssues, metadata, swaggerUrl);
+        } catch (ValidationException e) {
+            switch (e.getKey()) {
+                case WRONG_SERVICE_ID_KEY:
+                    foundNonConformanceIssues.put(REGISTRATION_PROBLEMS, e.getMessage());
+                    break;
+                case NO_METADATA_KEY:
+                    foundNonConformanceIssues.put(METADATA_PROBLEMS, e.getMessage());
+                    break;
+                default:
+                    foundNonConformanceIssues.put(CONFORMANCE_PROBLEMS, e.getMessage());
+            }
+            return generateBadRequestResponseEntity(e.getKey(), foundNonConformanceIssues);
         }
 
+        if (!foundNonConformanceIssues.isEmpty())
+            return generateBadRequestResponseEntity(NON_CONFORMANT_KEY, foundNonConformanceIssues);
+
         return new ResponseEntity<>("{\"message\":\"Service " + serviceId + " fulfills all checked conformance criteria\"}", HttpStatus.OK);
     }
 
+    private void validateSwaggerDocument(String serviceId, ConformanceProblemsContainer foundNonConformanceIssues, Map<String, String> metadata, String swaggerUrl) throws ValidationException {
+        if (swaggerUrl == null || swaggerUrl.equals("")) {
+            throw new ValidationException("Could not find Swagger Url", NON_CONFORMANT_KEY);
+        }
+
+        String swagger = verificationOnboardService.getSwagger(swaggerUrl);
+        AbstractSwaggerValidator swaggerParser;
+        swaggerParser = ValidatorFactory.parseSwagger(swagger, metadata, gatewayClient.getGatewayConfigProperties(), serviceId);
+
+        List<String> parserResponses = swaggerParser.getMessages();
+        if (parserResponses != null) foundNonConformanceIssues.put(CONFORMANCE_PROBLEMS, parserResponses);
+
+        Set<Endpoint> getMethodEndpoints = swaggerParser.getGetMethodEndpoints();
+        if (!getMethodEndpoints.isEmpty())
+            foundNonConformanceIssues.put(CONFORMANCE_PROBLEMS, verificationOnboardService.testGetEndpoints(getMethodEndpoints));
+
+        foundNonConformanceIssues.put(CONFORMANCE_PROBLEMS, verificationOnboardService.getProblemsWithEndpointUrls(swaggerParser));
```

Comment:
Are you missing an else here? The second one will override the entry always

Reply:
It doesn't overwrite it, instead it adds it to the container, Ill rename the function so its clearer at what it does.